packages/react/src/useEntityQuery.ts

```diff
@@ -1,23 +1,25 @@
-import { defineQuery } from "@latticexyz/recs";
+import { defineQuery, QueryFragment } from "@latticexyz/recs";
 import { useEffect, useMemo, useState } from "react";
+import { useDeepMemo } from "./utils/useDeepMemo";
+import isEqual from "fast-deep-equal";
+import { distinctUntilChanged, map } from "rxjs";
 
-type Args = Parameters<typeof defineQuery>;
-const defaultOptions: NonNullable<Args[1]> = { runOnInit: true };
+// This does a little more rendering than is necessary when arguments change,
+// but at least it's giving correct results now. Will optimize later!
```

Comment:
Should we open a follow up issue for this?

Reply:
I am not worried about it at the moment! If we get folks complaining about rerenders, then we can take another pass.